supervision/metrics/__init__.py

```diff
@@ -0,0 +1,2 @@
+from supervision.metrics.core import Metric, MetricTarget, UnsupportedMetricTargetError
```

Comment:
Do we need to expose the `UnsupportedMetricTargetError`?

Reply:
Yes. People need to know what to catch in their `with` statements. Catching a general `Exception` is bad practice.